fix(ui): make plugins pages responsive

aighita opened the pull request:
<!-- Thanks for submitting a Pull Request to Kestra. To help us review your contribution, please follow the guidelines below:

- Make sure that your commits follow the [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) specification e.g. `feat(ui): add a new navigation menu item` or `fix(core): fix a bug in the core model` or `docs: update the README.md`. This will help us automatically generate the changelog.
- The title should briefly summarize the proposed changes.
- Provide a short overview of the change and the value it adds.
- Share a flow example to help the reviewer understand and QA the change.
- Use "closes" to automatically close an issue. For example, `closes #1234` will close issue #1234. -->

### What changes are being made and why?

Closes #12752.
<!-- Please include a brief summary of the changes included in this PR e.g. closes #1234. -->

Plugins pages mobile responsive. Sidebar is not displayed on mobile. A hamburger icon is in turn displayed on the right side, not overlapping the docs.

---

### How the changes have been QAed?

https://github.com/user-attachments/assets/4b2b622c-6601-43c5-a34f-d5d3723b3fc6

Piyush-r-bhaskar commented on ui/src/components/plugins/PluginDocumentation.vue:
```diff
@@ -102,5 +102,89 @@
 </script>
 
 <style scoped lang="scss">
+    @import "@kestra-io/ui-libs/src/scss/variables";
```
remove changes from this file. We don't need this 🚀

aighita replied:
will do

Piyush-r-bhaskar commented on ui/src/components/plugins/PluginDocumentationWrapper.vue:
```diff
@@ -13,18 +13,42 @@
 </script>
 
 <style scoped lang="scss">
+@import "@kestra-io/ui-libs/src/scss/variables";
```
remove changes from this file.

aighita replied:
I need more context. Just the import or every change? I added the imports after commit https://github.com/kestra-io/kestra/pull/12829/commits/c7a859aaa7684c012593d1c7958772a73ca9529f because front-end tests failed.
Same for PluginList.vue and PluginUnified.vue?

Piyush-r-bhaskar commented on ui/src/components/plugins/PluginList.vue:
```diff
@@ -294,45 +294,53 @@
 </script>
 
 <style scoped lang="scss">
+@import "@kestra-io/ui-libs/src/scss/variables";
```
remove changes from this file.

Piyush-r-bhaskar commented on ui/src/components/plugins/PluginUnified.vue:
```diff
@@ -183,13 +183,15 @@
 </script>
 
 <style scoped lang="scss">
```
remove changes from this file.

Piyush-r-bhaskar commented on ui/src/components/docs/DocsLayout.vue:
```diff
@@ -1,15 +1,51 @@
+            @click="mobileMenuOpen = !mobileMenuOpen"
+            :aria-label="mobileMenuOpen ? 'Close menu' : 'Open menu'"
+        >
+            <svg
```
No direct svg Please use icons from https://pictogrammers.com/library/mdi/

You may take reference how project is importing at other place.

aighita replied:
Replaced with Menu and Close. See: DocsLayout.vue

Piyush-r-bhaskar reviewed (changes requested):
Good work on this @aighita . Thanks You 🚀

Few points to tackle before merge and you should be looking for correct file which is `/ui/src/components/plugins/Plugin.vue`

- Handle this bg-color difference.

![Image](https://github.com/user-attachments/assets/d4161dc8-7e6a-4995-be47-7ee24eff622a)

- I would love to fix the title and hamburger and scroll should start from below that and also try to make the overflowing ellipsis.

![Image](https://github.com/user-attachments/assets/5d9a65ca-dde9-47b7-92a2-f94209c68f3c)

- Similar as above make the search and close button fix as header and scroll should start from below.

<img width="862" height="1724" alt="Image" src="https://github.com/user-attachments/assets/ed2451b2-3a03-4f52-9100-d756c991113f" />

- I see the Hamburger is appearing inside Right Context bar Doc / News..Need to remove

<img width="1334" height="1724" alt="Image" src="https://github.com/user-attachments/assets/a094f61c-74d5-40b6-a08b-3d02eee5867b" />

Piyush-r-bhaskar commented on ui/src/components/docs/DocsLayout.vue:
```diff
@@ -1,15 +1,51 @@
-                <span class="toggle-btn text-body-tertiary">{{ $t((collapsed ? 'open' : 'close') + ' sidebar').toUpperCase() }}</span>
-            </div>
-            -->
+        <button 
```
You may use icon type `el-button`

aighita replied:
Replaced button with el-button. See: DocsLayout.vue

aighita commented:
* Replaced button with el-button.
* Replaced SVG with https://pictogrammers.com/library/mdi/.
* Removed changes from requested files.
* Handled bg color difference.
* Fixed title and menu icon.
* Scroll bar start from below the title and menu icon header.
* Conditionally render the hamburger menu and close buttons on plugins route as they are used in DocsLayout.vue but that component might be used in other parts too. Wasn't quite clear for me where exactly Right Context bar Doc / News.. was in the picture you provided so I see this the best fix as that menu is only used in the plugins page.

As a result of these changes desktop styles were affected too: title is now fixed on scroll on any device. Scroll bar starts under the title container.

QA

https://github.com/user-attachments/assets/73e31367-6e9f-4a7a-9271-306be3117662

Piyush-r-bhaskar reviewed (changes requested):
@aighita  I see your changes are applied on large screen as well. could you inly make changes which target small screens

and keep the icon size as it is.

aighita commented:
@Piyush-r-bhaskar 

Title + Release Notes shouldn't be fixed/sticky on large screen?

Icon size keep as is meaning revert to it's original size before my changes? 

Which icon are you referring to? From plugin title (ex in the video: Docker icon) or hamburger icon?

Piyush-r-bhaskar commented:
Yes , should not be sticky on large screen.

and I mean Plugin icon/image beside title.

aighita commented:
@Piyush-r-bhaskar

MilosPaunovic commented:
Hey @Piyush-r-bhaskar, can you check this one again when you get the chance?

Piyush-r-bhaskar committed:
fix(ui): make plugins pages responsive (#12752)

Piyush-r-bhaskar committed:
Merge branch 'develop' into fix/plugins-pages-#12752

Piyush-r-bhaskar committed:
fix(core): tweaks to make it little better

Piyush-r-bhaskar reviewed (approved):
LGTM 🚀. Thank you for your contribution @aighita  🎉
